src/isl/isl_constraint.rs

```diff
@@ -46,6 +48,14 @@ impl IslConstraint {
         IslConstraint::OrderedElements(isl_types.into())
     }
 
+    /// Creates a [IslConstraint::Fields] using the field names and [IslTypeRef]s referenced inside it
+    pub fn fields<I>(fields: I) -> IslConstraint
+    where
+        I: Iterator<Item = (String, IslTypeRef)>,
```

Comment:
question—is `fields(fields: Iterator<(String, IslTypeRef)>)` illegal in Rust or is this a stylistic choice?

Reply:
A stylistic choice to using `trait`s as a parameter to function.